### PR TITLE
feat(logger): add get_correlation_id method

### PR DESCRIPTION
**Pull request by michaelbrewer**
**Issue #, if available:**

- https://github.com/awslabs/aws-lambda-powertools-python/issues/518

## Description of changes:

- Call `remove_keys["correlation_id"]` when the correlation_id is set to None.
- Add `get_correlation_id` call

```python
from aws_lambda_powertools import Logger

logger = Logger()
logger.set_correlation_id("Foo")
assert "Foo" == logger.get_correlation_id()
logger.set_correlation_id(None)
assert logger.get_correlation_id() is None
```

**Checklist**

 

* [X] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [X] Update tests
* [X] Update docs
* [X] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Nr18 — APPROVED** [review]
LGTM ;-)

**Comment by michaelbrewer** [comment]
@heitorlessa should we go ahead with this feature?

**Comment by heitorlessa** [comment]
I've just added a comment in the RFC.

TL;DR: Definitely yes for Getter, perhaps even a generic one to grab any key. Not yet for the remove correlation ID method.

File: aws_lambda_powertools/logging/logger.py [hunk 1]
```diff
@@ -387,15 +387,28 @@ def structure_logs(self, append: bool = False, **keys):
...
         Parameters
         ----------
-        value : str
+        value : str, optional
+            Value for the correlation id. None will remove the correlation_id
+        """
+        if value is None:
+            self.remove_keys(["correlation_id"])
+        else:
+            self.append_keys(correlation_id=value)
```
Comment (heitorlessa) [inline aws_lambda_powertools/logging/logger.py:401]: Keys with `None` value get removed by default when emitting a log message, so this wouldn't be necessary: https://github.com/awslabs/aws-lambda-powertools-python/blob/89e8151b5c930ba79d685d05a83ae362cf94839d/aws_lambda_powertools/logging/formatter.py#L126

**Review by heitorlessa — CHANGES_REQUESTED** [review]
By default, Logger [removes any key with `None` value before emitting a log entry](https://github.com/awslabs/aws-lambda-powertools-python/blob/89e8151b5c930ba79d685d05a83ae362cf94839d/aws_lambda_powertools/logging/formatter.py#L126).

File: tests/functional/test_logger.py [hunk 2]
```diff
@@ -460,6 +460,32 @@ def handler(event, _):
...
+    logger = Logger(service=service_name, stream=stdout)
+    logger.set_correlation_id("foo")
+
+    # WHEN calling set_correlation_id with None
+    logger.set_correlation_id(None)
+
+    # THEN there should be no correlation_id
+    logger.info("Foo")
+    log = capture_logging_output(stdout)
+    assert "correlation_id" not in log
```
Comment (heitorlessa) [inline tests/functional/test_logger.py:474]: That's not needed ;) https://github.com/awslabs/aws-lambda-powertools-python/blob/89e8151b5c930ba79d685d05a83ae362cf94839d/tests/functional/test_logger_powertools_formatter.py#L175

File: aws_lambda_powertools/logging/logger.py (hunk 1 above)
Comment (michaelbrewer) [reply aws_lambda_powertools/logging/logger.py]: resolved.

File: tests/functional/test_logger.py (hunk 2 above)
Comment (michaelbrewer) [reply tests/functional/test_logger.py]: done

**Comment by michaelbrewer** [comment]
@heitorlessa  made the changes. the diff is much more minimal, but still meet the needs of the original RFC

**Review by heitorlessa — APPROVED** [review]

**Comment by heitorlessa** [comment]
Thanks a lot again @michaelbrewer !
